analyzer.rb

```diff
@@ -159,16 +159,28 @@ def build_commands(config, platform, project_type_filter, id_filters = nil)
   def build_test_commands(config, platform, project_type_filter)
     configuration = "#{config}|#{platform}"
     build_commands = []
+    errors = []
 
     @solution[:projects].each do |project|
-      next unless project[:mappings]
-
-      project_configuration = project[:mappings][configuration]
-      next unless project_configuration
-
       # Check whether it is a UITest project
+      # Do this check as soon as possible,
+      # to allow collecting errors of building test command
+      # for relevant (UITest) projects.
       next if project[:tests].nil? || !project[:tests].include?(Tests::UITEST)
 
+      test_project = project[:name]
+
+      unless project[:mappings]
+        errors << "#{test_project} not found in solution mappings\n#{project}"
```

Comment:
here we should add it as multiple array elements, otherwise it will be hard to read. Either with concat [] or with multiple <<s